draft-ietf-teep-protocol.cddl

```diff
@@ -99,6 +102,8 @@ query-response = [
   options: {
     ? token => bstr .size (8..64),
     ? selected-teep-cipher-suite => $teep-cipher-suite,
+    ? supported-freshness-mechanisms => [ + $freshness-mechanism ],
+    ? challenge => bstr .size (8..512),
```

Comment:
I don't see why these are needed in the QueryResponse, they should be in an Error message instead.

Reply:
I agree with Dave. Move them to the Error message

Reply:
@dthaler @hannestschofenig 
Moved them in the Error message.
Also split the challenge to challenge-tam and challenge-agent.
Thanks!